src/cpu.cst

```diff
@@ -1,51 +1,41 @@
+IO_LOC  "output_peripherals[3]" 30;
+IO_PORT "output_peripherals[3]" DRIVE=8 IO_TYPE=LVCMOS33;
 
-IO_LOC  "input_peripherals[0]" 25;
-IO_PORT "input_peripherals[0]" DRIVE=8 IO_TYPE=LVCMOS33 PULL_MODE=DOWN;
+IO_LOC  "output_peripherals[2]" 29;
+IO_PORT "output_peripherals[2]" DRIVE=8 IO_TYPE=LVCMOS33;
 
 IO_LOC  "input_peripherals[1]" 26;
-IO_PORT "input_peripherals[1]" DRIVE=8 IO_TYPE=LVCMOS33 PULL_MODE=DOWN;
+IO_PORT "input_peripherals[1]" IO_TYPE=LVCMOS33 PULL_MODE=DOWN;
 
-IO_LOC  "input_peripherals[2]" 3;
+IO_LOC  "input_peripherals[2]" 80;
 IO_PORT "input_peripherals[2]" IO_TYPE=LVCMOS18;
 
 IO_LOC  "input_peripherals[3]" 4;
 IO_PORT "input_peripherals[3]" IO_TYPE=LVCMOS18;
 
-IO_LOC  "output_peripherals[0]" 27;
-IO_PORT "output_peripherals[0]" DRIVE=8 IO_TYPE=LVCMOS33;
-
 IO_LOC  "output_peripherals[1]" 28;
 IO_PORT "output_peripherals[1]" DRIVE=8 IO_TYPE=LVCMOS33;
 
-IO_LOC  "output_peripherals[2]" 15;
-IO_PORT "output_peripherals[2]" DRIVE=8 IO_TYPE=LVCMOS18;
-
-IO_LOC  "output_peripherals[3]" 16;
-IO_PORT "output_peripherals[3]" DRIVE=8 IO_TYPE=LVCMOS18;
-
 IO_LOC  "clock" 52;
 IO_PORT "clock" IO_TYPE=LVCMOS33 PULL_MODE=UP;
 
-IO_LOC  "led[0]" 10;
+IO_LOC  "output_peripherals[0]" 10;
+IO_PORT "output_peripherals[0]" DRIVE=8 IO_TYPE=LVCMOS18;
+
+IO_LOC  "led[0]" 11;
 IO_PORT "led[0]" DRIVE=8 IO_TYPE=LVCMOS18;
 
-IO_LOC  "led[1]" 11;
+IO_LOC  "led[1]" 13;
 IO_PORT "led[1]" DRIVE=8 IO_TYPE=LVCMOS18;
 
-IO_LOC  "led[2]" 13;
+IO_LOC  "led[2]" 14;
 IO_PORT "led[2]" DRIVE=8 IO_TYPE=LVCMOS18;
 
-IO_LOC  "led[3]" 14;
+IO_LOC  "led[3]" 15;
 IO_PORT "led[3]" DRIVE=8 IO_TYPE=LVCMOS18;
 
-IO_LOC  "flashClk" 59;
-IO_PORT "flashClk" DRIVE=8 IO_TYPE=LVCMOS33 PULL_MODE=UP;
-
-IO_LOC  "flashMiso" 62;
-IO_PORT "flashMiso" DRIVE=8 IO_TYPE=LVCMOS33 PULL_MODE=UP;
-
-IO_LOC  "flashMosi" 61;
-IO_PORT "flashMosi" DRIVE=8 IO_TYPE=LVCMOS33 PULL_MODE=UP;
+IO_LOC  "led[4]" 16;
+IO_PORT "led[4]" DRIVE=8 IO_TYPE=LVCMOS18;
 
-IO_LOC  "flashCs" 60;
-IO_PORT "flashCs" DRIVE=8 IO_TYPE=LVCMOS33 PULL_MODE=UP;
+IO_LOC  "input_peripherals[0]" 3;
```

Comment:
Didin't get why scrambling around the pins so much :thinking:

Reply:
Gowin was complaining about unused pins and also about some properties that were apparently invalid :confused: